### DISCUSSION TITLE
default gateway Information / AS Information  in ansible inventory

### DISCUSSION THREAD

**Discussion by Joshi425** [netbox-community/netbox#7482] [Q&A]
We use a spine leaf setup in our dc and some maschines need to be aware of the leaf they are in. To accomplish this I wanted to create locations with the racks as children or just assign a config context to the racks in each leaf. But neither is supported yet.

What is the best way to model this information in netbox so I can access it in the ansible Inventory? Just tagging them feels wrong.

**Comment by candlerb** [comment]
> I wanted to create locations with the racks as children

This is exactly how it works today - each Rack can be assigned to a parent Location.  In fact, locations used to be called "rack groups" in older versions of Netbox.  Can you explain what you need that's different?

> or just assign a config context to the racks in each leaf

Personally, I think it's a reasonable feature request to want to be able to assign config contexts to Racks and to Locations.  I was surprised it wasn't already there.

**Reply by jeremystretch** [reply]
> Personally, I think it's a reasonable feature request to want to be able to assign config contexts to Racks

I explain my reasoning [here](https://github.com/netbox-community/netbox/issues/4699#issuecomment-636182283). If someone can raise a use case in which they need to configure a device specifically because of the rack it's in, rather than a function for which the rack has been designated, I'd be very interested to hear it.

**Reply by Joshi425** [reply]
> 
> 
> > I wanted to create locations with the racks as children
> 
> This is exactly how it works today - each Rack can be assigned to a parent Location. In fact, locations used to be called "rack groups" in older versions of Netbox. Can you explain what you need that's different?

I want to access information like this in the ansible Inventory. The information rendered by the inventory plugin doesn't include custom fields in racks or someting similar. 

It would also be fine to attach information like the AS numbers to the IP Range the devices primary IPv4 address is in, but ipranges also don't support Config Contexts.

**Reply by cpmills1975** [reply]
You can expose information held within the rack object by querying NetBox in a task using the nb_lookup plug-in for Ansible.

The device in your inventory will have a reference to its rack. Within a task, use the nb_lookup plug-in to obtain all the information for that rack which can be stored for later use using a set_fact task or directly within the task using task vars.

I use this technique extensively - a web hook get fired in to my automation stack for updates to a device, but I want to know what power outlets it's plugged in to. The webhook payload doesn't contain this info, so I use nb_lookup to query all connected power ports for the device id that's been submitted and then act upon that within my Ansible based automation tooling.
